flink-runtime/src/main/java/org/apache/flink/runtime/state/StateSnapshotTransformer.java

```diff
@@ -183,4 +207,17 @@ public MapStateSnapshotTransformer(StateSnapshotTransformer<V> entryValueTransfo
 
 		Optional<StateSnapshotTransformer<byte[]>> createForSerializedState();
 	}
+
+	abstract class StateSnapshotTransformFactoryWrapAdaptor<S, T> implements StateSnapshotTransformFactory<T> {
```

Comment:
Maybe `CollectionStateSnapshotTransformerFactory` would be a better name

Reply:
at this point, there is nothing about collections yet, just wrapping some other factory. I can override `createForDeserializedState` as well to make it a true adaptor.

Reply:
But maybe we could find a bit more meaningful name for this class.